plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1146,6 +1149,47 @@ public void renameTable(ConnectorSession session, ConnectorTableHandle tableHand
         catalog.renameTable(session, ((IcebergTableHandle) tableHandle).getSchemaTableName(), newTable);
     }
 
+    @Override
+    public void setTableProperties(ConnectorSession session, ConnectorTableHandle tableHandle, Map<String, Optional<Object>> properties)
+    {
+        IcebergTableHandle table = (IcebergTableHandle) tableHandle;
+        BaseTable icebergTable = (BaseTable) catalog.loadTable(session, table.getSchemaTableName());
+
+        transaction = icebergTable.newTransaction();
+        UpdateProperties updateProperties = transaction.updateProperties();
+
+        for (Map.Entry<String, Optional<Object>> propertyEntry : properties.entrySet()) {
+            String trinoPropertyName = propertyEntry.getKey();
+            Optional<Object> propertyValue = propertyEntry.getValue();
+
+            switch (trinoPropertyName) {
+                case FILE_FORMAT_PROPERTY:
+                    updateProperty(updateProperties, DEFAULT_FILE_FORMAT, propertyValue, format -> ((IcebergFileFormat) format).toIceberg().toString());
+                    break;
+                case FORMAT_VERSION_PROPERTY:
+                    updateProperty(updateProperties, FORMAT_VERSION, propertyValue, formatVersion -> Integer.toString((int) formatVersion));
```

Comment:
Is it all that's needed here?

i would expect

- a failure when user tries to downgrade a v2 table with deletion files to the v1 version; this might be possible only for fully compacted tables (or not at all, idk)
- for v1 -> v2: `TableMetadata.upgradeToFormatVersion(2)`. Is it triggered implicitly?

Reply:
> a failure when user tries to downgrade a v2 table with deletion files to the v1 version; this might be possible only for fully compacted tables (or not at all, idk)

This fails with an exception "Cannot downgrade v2 table to v1" , I'll add a test and should probably wrap in a TrinoException

Reply:
Sure, added